setup.py

```diff
@@ -24,7 +41,7 @@
 
 setup(
     name='superflore',
-    version='0.3.0',
+    version=get_version_from_git_describe(),
```

Comment:
@tfoote does this disrupt anything for releases?

Reply:
The step that previously created a commit to change the value of `version` is replaced by one that adds a tag. @allenh1 < Would this be too much of a disruption?

Reply:
I've dropped this controversial commit from the PR and added various other improvements instead. Please re-review.

Reply:
Since I intend to someday propose an alternative implementation that addresses Tully's objections, I now think it would make more sense to create a new PR for the bitbake generator changes and just have the setup.py commit be in this PR.